Modify EL batching to doc-wise streaming approach (#12367)
explosion/spaCy: website/docs/api/inmemorylookupkb.mdx

## InMemoryLookupKB.get_candidates {id="get_candidates",tag="method"}

Given textual mentions for an arbitrary number of documents as input, retrieve a
list of candidate entities of type [`InMemoryCandidate`](/api/kb#candidate) for
each mention. The [`EntityLinker`](/api/entitylinker) component passes a

> generator yielding all mentions to retreive candidates for as
> [`SpanGroup`](/api/spangroup)) per document. The decision of how to batch

Reviewer: ```suggestion
generator that yields mentions as [`SpanGroup`](/api/spangroup))s per document. The decision of how to batch
```

> from spacy.tokens import SpanGroup
> nlp = English()
> doc = nlp("Douglas Adams wrote 'The Hitchhiker's Guide to the Galaxy'.")
> candidates = kb.get_candidates([SpanGroup(doc, spans=[doc[0:2], doc[3:]]])
> ```

| Name        | Description                                                                                                                                                                       |
| ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `mentions`  | The textual mention or alias. ~~Iterable[SpanGroup]~~                                                                                                                             |

Author: Docs updated in https://github.com/explosion/spaCy/pull/12367/commits/638103eb1b6a05e24c0dea9eb2d925726945a738.
